### PR TITLE
Add Robust Error Checking to Library and Worker's Communication Patterns

### PR DESCRIPTION
**Pull request by tphung3**
## Proposed Changes

This PR introduces more error checking in the worker when a library first starts up. A worker now sanity-checks a library's name, task_id, and exec_mode instead of just the library's name.

A new library protocol number is also added and should be incremented every time the library changes its communication pattern with the worker.

Issue: https://github.com/cooperative-computing-lab/cctools/issues/3439.

## Merge Checklist

The following items must be completed before PRs can be merge.
Check these off to verify you have completed all steps.

- [x] `make test`       Run local tests prior to pushing.
- [x] `make format`     Format source code to comply with lint policies. Note that some lint errors can only be resolved manually (e.g., Python)
- [x] `make lint`       Run lint on source code prior to pushing.
- [x] Manual Update     Update the manual to reflect user-visible changes.
- [x] Type Labels       Select a github label for the type: bugfix, enhancement, etc.
- [x] Product Labels    Select a github label for the product: TaskVine, Makeflow, etc.
- [x] PR RTM            Mark your PR as ready to merge.

### CODE REVIEW

**Review by dthain — CHANGES_REQUESTED** [review]

File: taskvine/src/manager/vine_manager.c [hunk 1]
```diff
@@ -1461,9 +1461,10 @@ static vine_msg_code_t handle_taskvine(struct vine_manager *q, struct vine_worke
 {
 	char items[4][VINE_LINE_MAX];
 	int worker_protocol;
+	int worker_library_protocol;
 
-	int n = sscanf(line, "taskvine %d %s %s %s %s", &worker_protocol, items[0], items[1], items[2], items[3]);
-	if (n != 5)
+	int n = sscanf(line, "taskvine %d %d %s %s %s %s", &worker_protocol, &worker_library_protocol, items[0], items[1], items[2], items[3]);
```
Comment (btovar) [inline taskvine/src/manager/vine_manager.c:1466]: I'm not sure about having a different number for the library protocol.  I'd keep only the worker_protocol and increment it if the library communication between the worker and library changes.

File: taskvine/src/manager/vine_protocol.h [hunk 2]
```diff
@@ -13,7 +13,9 @@ worker, and catalog, but should not be visible to the public user API.
 #ifndef VINE_PROTOCOL_H
 #define VINE_PROTOCOL_H
 
-#define VINE_PROTOCOL_VERSION 11
+#define VINE_PROTOCOL_VERSION 12
+
+#define VINE_LIBRARY_PROTOCOL_VERSION 1
```
Comment (btovar) [inline taskvine/src/manager/vine_protocol.h:18]: See above, this sounds like an over complication.

**Review by btovar — CHANGES_REQUESTED** [review]

**Review by btovar — APPROVED** [review]

**Comment by btovar** [comment]
ready to merge?

**Comment by tphung3** [comment]
Yep. It's RTM.

**Pull request merged**